x-pack/plugins/enterprise_search/server/routes/app_search/engines.ts

```diff
@@ -9,38 +9,52 @@ import { schema } from '@kbn/config-schema';
 
 import { ENGINES_PAGE_SIZE } from '../../../common/constants';
 
-export function registerEnginesRoute({ router, config }) {
+export function registerEnginesRoute({ router, config, log }) {
   router.get(
     {
       path: '/api/app_search/engines',
       validate: {
         query: schema.object({
-          type: schema.string(),
+          type: schema.oneOf([schema.literal('indexed'), schema.literal('meta')]),
           pageIndex: schema.number(),
         }),
       },
     },
     async (context, request, response) => {
-      const appSearchUrl = config.host;
-      const { type, pageIndex } = request.query;
-
-      const url = `${appSearchUrl}/as/engines/collection?type=${type}&page[current]=${pageIndex}&page[size]=${ENGINES_PAGE_SIZE}`;
-      const enginesResponse = await fetch(url, {
-        headers: { Authorization: request.headers.authorization },
-      });
-
-      if (enginesResponse.url.endsWith('/login')) {
-        return response.ok({
-          body: { message: 'no-as-account' },
-          headers: { 'content-type': 'application/json' },
+      try {
+        const appSearchUrl = config.host;
+        const { type, pageIndex } = request.query;
+
+        const url = `${appSearchUrl}/as/engines/collection?type=${type}&page[current]=${pageIndex}&page[size]=${ENGINES_PAGE_SIZE}`;
+        const enginesResponse = await fetch(url, {
+          headers: { Authorization: request.headers.authorization },
         });
-      }
 
-      const engines = await enginesResponse.json();
-      return response.ok({
-        body: engines,
-        headers: { 'content-type': 'application/json' },
-      });
+        if (enginesResponse.url.endsWith('/login')) {
```

Comment:
It's surprising to me that the `url` prop here would end with `/login` in case of an invalid account. Does App Search redirect to `/login` here when an invalid auth header is provided? Calling the API directly would return a `401` status code.

Reply:
Yep, this is on App Search's end - I believe our Ruby logic is to default to the /login page no matter what if we encounter any invalid auth (even when hitting API endpoints).

Reply:
To be clear I don't 100% agree with it (as someone who approaches things from a very RESTful API/separate your front-end & back-end view :), but I also would have no idea how to change it, so probably out of scope for now.